router/data_store.go

```diff
@@ -39,196 +33,206 @@ type SyncHandler interface {
 	Remove(id string) error
 }
 
-func NewEtcdDataStore(etcd EtcdClient, prefix string) DataStore {
-	return &etcdDataStore{
-		prefix:   prefix,
-		etcd:     etcd,
-		stopSync: make(chan bool),
-	}
+type pgDataStore struct {
+	pgx *pgx.ConnPool
+
+	rt string
+
+	doneo sync.Once
+	donec chan struct{}
 }
 
-type etcdDataStore struct {
-	prefix   string
-	etcd     EtcdClient
-	stopSync chan bool
+// NewPostgresDataStore returns a DataStore that stores route information in a
+// Postgres database. It uses pg_notify and a listener connection to watch for
+// route changes.
+func NewPostgresDataStore(routeType string, pgx *pgx.ConnPool) *pgDataStore {
+	return &pgDataStore{
+		pgx:   pgx,
+		rt:    routeType,
+		donec: make(chan struct{}),
+	}
 }
 
-var ErrExists = errors.New("router: route already exists")
-var ErrNotFound = errors.New("router: route not found")
+var sqlAddRoute = `INSERT INTO routes (parent_ref, type, config) VALUES ($1, $2, $3::json) RETURNING route_id, created_at, updated_at`
```

Comment:
yeah, I noticed the same and made the change in my branch. Feel free to change here too if you want.